apps/transport/lib/transport/history.ex

```diff
@@ -44,20 +44,21 @@ defmodule Transport.History do
   defp needs_to_be_updated(resource) do
     backuped_resources = get_already_backuped_resources(resource)
 
-    max_last_modified =
-      backuped_resources
-      |> Enum.map(fn r ->
-        {r.updated_at, r.content_hash}
-      end)
-      |> Enum.max_by(fn {date, _hash} -> date end, fn -> nil end)
-
-    case max_last_modified do
-      nil ->
-        true
-
-      {max_last_modified, content_hash} ->
-        modification_date = modification_date(resource)
-        max_last_modified < modification_date && content_hash != resource.content_hash
+    if length(backuped_resources) == 0 do
+      true
+    else
+      already_there = backuped_resources
+      |> Enum.find(fn r -> r.content_hash == resource.content_hash end)
```

Comment:
I _think_ if there is only one element, put it in the function call instead of `|>`

Reply:
And I’d use again an Enum.empty? instead of comparing against nil

Reply:
hum I don't get how you want to use Enum.empty? here.
the Enum.find function return nil is not found

Reply:
Yup, sorry